keras_nlp/layers/transformer_encoder.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TransformerEncoder class definition."""
+
+from tensorflow import keras
+
+from keras_nlp.layers.transformer_utils import merge_padding_and_additional_mask
+
+
+class TransformerEncoder(keras.layers.Layer):
+    """Transformer encoder.
+
+    This class follows the architecture of transformer encoder layer in paper
+    "Attention is All You Need" (https://arxiv.org/abs/1706.03762). Users can
+    instantiate multiple instances of this class to stack up the encoder.
+
+    Args:
+        feedforward_dim: int, the hidden size of feedforward network.
+        num_heads: int, the number of heads in MultiHeadAttention.
+        dropout: float, defaults to 0. the dropout value, shared by
+            MultiHeadAttention and feedforward network.
+        activation: string or `tf.keras.activations`, defaults to "relu". the
+            activation function of feedforward network.
+        layer_norm_epsilon: float, defaults to 1e-5. The eps value in layer
+            normalization components.
+        name: string, defaults to "TransformerDecoder". The name of the layer.
+        **kwargs: other keyword arguments.
+
+    Examples:
+
+    ```python
+    # Create a single transformer decoder layer.
+    encoder = TransformerEncoder(feedforward_dim=64, num_heads=8)
```

Comment:
should we do this with the full import? keras_nlp.layer.TransformerEncoder ? That would follow the norm for code examples on core keras

Reply:
SG!